### ISSUE TITLE
SNOW-257769: Support processing multi-column result set in multi statement execution

### ISSUE THREAD

**Issue by adatzer** [snowflakedb/gosnowflake#365]
### Issue description
What happens:
Selecting more than one column in a multi-statement query, executing with `db.QueryContext()` and processing the result sets,  gives `panic: runtime error: index out of range [1] with length 1`. 
Also, `rows.Columns()` gives only `[multiple statement execution]` in all result sets.

What should happen:
Ideally, process the multiple result sets as with a single statement query.
Alternatively, a controlled error.

### Example code
```go
import (
	"context"
	"database/sql"
        sf "github.com/snowflakedb/gosnowflake"
        "log"
)

// ...

// assuming table test_tbl exists with int columns c1,c2
multi_stmt_query := "select c1 from test_tbl;\n" +
		"select c1,c2 from test_tbl;"

ctx, _ := sf.WithMultiStatement(context.Background(), 0)

rows, err := db.QueryContext(ctx, multi_stmt_query)
if err != nil {
	t.Fatalf("failed to query multiple statements: %v", err)
}
defer rows.Close()

for rows.Next() {
        log.Printf("placeholder")
}
for  rows.NextResultSet() {
	for rows.Next() {
                log.Printf("placeholder")
        }	
}
```

### Error log
```
panic: runtime error: index out of range [1] with length 1

goroutine 22 [running]:
github.com/snowflakedb/gosnowflake.(*snowflakeRows).Next(0xc000212180, 0xc00031c1e0, 0x1, 0x1, 0xc0005bc000, 0x22)
	/home/adatzer/go/pkg/mod/github.com/snowflakedb/gosnowflake@v1.3.12/rows.go:173 +0x3db
database/sql.(*Rows).nextLocked(0xc000158580, 0xc0000e0000)
	/usr/local/go/src/database/sql/sql.go:2858 +0xbf
database/sql.(*Rows).Next.func1()
	/usr/local/go/src/database/sql/sql.go:2836 +0x3c
database/sql.withLock(0xe70d60, 0xc0001585b0, 0xc0007c77a0)
	/usr/local/go/src/database/sql/sql.go:3284 +0x69
database/sql.(*Rows).Next(0xc000158580, 0xd)
	/usr/local/go/src/database/sql/sql.go:2835 +0x87

```

### Configuration
*Driver version (or git SHA):* v1.3.12
*Go version:* go version go1.15.5 linux/amd64
*Server version:* 4.43.2
*Client OS:* Ubuntu 20.10

**Comment by sfc-gh-jbahk** [comment]
@adatzer This has been addressed and will be available on the next release, v1.4.1.

**Comment by adatzer** [comment]
Thank you very much
